operator/backrest/restore.go

```diff
@@ -424,12 +424,10 @@ func CreateRestoredDeployment(restclient *rest.RESTClient, cluster *crv1.Pgclust
 		return err
 	}
 
-	cluster.Spec.UserLabels[config.LABEL_CURRENT_PRIMARY] = restoreToName
-
```

Comment:
I'm guessing we don't need to  update the annotation here?

Reply:
not currently, because of the updates to `util.PatchClusterCRD`, which now takes an additional string value for the current primary annotation, deployment annotation, and PrimaryStorage value (shown below)
```
// store the workflowID in a user label
	cluster.Spec.UserLabels[crv1.PgtaskWorkflowID] = workflowID
	// patch the pgcluster CRD with the updated info
	if err = util.PatchClusterCRD(restclient, cluster.Spec.UserLabels, cluster, restoreToName, namespace); err != nil {
		log.Error("could not patch primary crv1 with labels")
		return err
	}
```
The reason behind this change is to keep the all the indications of the current primary consistent after failover or restoration events where previously these would result in the pgcluster CRD becoming 'out of sync.'  With the upgrade procedure treating the CRD as the source of truth, it has become more important that these values are kept up to date.